### PR TITLE
[Exam] Block Instructors from registering as students to exam

### LINKED ISSUE #2489: [Enhancement] Block instructors of a course to be registered as participants in the course's exams
### Is your feature request related to a problem?
Instructors have access to sensitive information about the exam. As instructors can try out the exam by using the test run feature, they should not be able to register as a student to the exam. The current approach also creates additional overhead for the developers. This is because currently, we ignore all the instructor submissions for the exam scores page and assessment statistics which can be confusing during testing if the devs participated in an exam with their admin/instructor account.

### Describe the solution you'd like
Limit the users which can be registered to the exam to only users who are not instructors in the specific course.

### Additional context
We came across this issue a few times during testing where we thought we introduced a bug, only for it to be due to our participations being filtered out, _as expected_.

### PR DESCRIPTION
**Pull request by anditurdiu**
### Checklist
- [x] I tested *all* changes and *all* related features with different users (student, tutor, instructor, admin) on the test server https://artemistest.ase.in.tum.de.
- [x] Server: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/server.html).
- [x] Server: I added multiple integration tests (Spring) related to the features (with a high test coverage)
- [x] Server: I implemented the changes with a good performance and prevented too many database calls
- [x] Client: I followed the [coding and design guidelines](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/client.html).

### Motivation and Context
Developers and testers often run into unexpected behaviour which is a result of instructors signing up as students to the exam. This breaks exam & test run related features. Therefore we want to block instructors of the course and administrators from signing up to an exam. 

Fixes #2489

### Description
Throw forbidden error on the client if the user tries to register an instructor or administrator to the exam. 

### Steps for Testing
1. Log in to Artemis
2. Navigate to an exam
3. Try to register an instructor of the course or an administrator to the exam
4. A forbidden error should be thrown

### Test Coverage
ExamResource: 99%

### CODE REVIEW

File: src/main/webapp/i18n/de/exam.json [hunk 1]
```diff
@@ -57,6 +57,9 @@
                 "numberOfExercisesInExamNotSet": "Die Anzahl an Aufgaben in der Klausur ist nicht gesetzt.",
                 "tooFewExerciseGroups": "Die Anzahl der Aufgabengruppen ist zu klein.",
                 "tooManyMandatoryExerciseGroups": "Die Anzahl verpflichtender Aufgabengruppen ist zu groß."
+            },
+            "error": {
+                "cannotRegisterInstructor": "Du kannst keine Instruktoren und Administratoren zu Klausuren anmelden."
```
Comment (krusche) [inline src/main/webapp/i18n/de/exam.json:62]:
```suggestion
                "cannotRegisterInstructor": "Du kannst keine Dozenten oder Administratoren zu Klausuren anmelden."
```

File: src/main/java/de/tum/in/www1/artemis/web/rest/ExamResource.java [hunk 2]
```diff
@@ -411,6 +411,9 @@ public ExamResource(UserService userService, CourseService courseService, ExamRe
         if (student.isEmpty()) {
             return notFound();
         }
+        if (student.get().getGroups().contains(exam.getCourse().getInstructorGroupName()) || authCheckService.isAdmin(student.get())) {
+            return forbidden("exam", "cannotRegisterInstructor", "You cannot register instructors and administrators to exams.");
```
Comment (krusche) [inline src/main/java/de/tum/in/www1/artemis/web/rest/ExamResource.java:415]:
```suggestion
            return forbidden("exam", "cannotRegisterInstructor", "You cannot register instructors or administrators to exams.");
```

File: src/main/webapp/i18n/en/exam.json [hunk 3]
```diff
@@ -57,6 +57,9 @@
                 "numberOfExercisesInExamNotSet": "The number of exercises in the exam is not set.",
                 "tooFewExerciseGroups": "The number of exercise groups is too small.",
                 "tooManyMandatoryExerciseGroups": "The number of mandatory exercise groups is too large."
+            },
+            "error": {
+                "cannotRegisterInstructor": "You cannot register instructors and administrators to exams."
```
Comment (krusche) [inline src/main/webapp/i18n/en/exam.json:62]:
```suggestion
                "cannotRegisterInstructor": "You cannot register instructors or administrators to exams."
```

**Review by krusche — APPROVED** [review]
code looks good (I did not tested it)

**Review by EneaGore — DISMISSED** [review]
Warning message and console output look great.

**Review by airelawaleria — APPROVED** [review]

**Review by JohannesWeiss — DISMISSED** [review]

**Review by simonzilker — DISMISSED** [review]

**Review by krusche — APPROVED** [review]

**Pull request merged**